### PR TITLE
[SPARK-17808][PYSPARK] Upgraded version of Pyrolite to 4.13

### PR DESCRIPTION
**Pull request by BryanCutler**
## What changes were proposed in this pull request?

Upgraded to a newer version of Pyrolite which supports serialization of a BinaryType StructField for PySpark.SQL
## How was this patch tested?

Added a unit test which fails with a raised ValueError when using the previous version of Pyrolite 4.9 and Python3

### CODE REVIEW

File: python/pyspark/sql/tests.py [hunk 1]
```diff
@@ -1702,6 +1702,14 @@ def test_cache(self):
             "does_not_exist",
             lambda: spark.catalog.uncacheTable("does_not_exist"))
 
+    def test_BinaryType_serialization(self):
+        # Pyrolite version <= 4.9 could not serialize BinaryType with Python3 SPARK-17808
+        schema = StructType([StructField('mybytes', BinaryType())])
+        data = [[bytearray(b'here is my data')],
+                [bytearray(b'and here is some more')],]
```
Comment (holdenk) [inline python/pyspark/sql/tests.py]: linter error here with the ','

**Comment by holdenk** [comment]
Thanks for working on this - the pylint script found a style problem (PEP8 checks failed.
./python/pyspark/sql/tests.py:1709:54: E231 missing whitespace after ',') - if you want to test the style locally first you can use ./dev/lint-python

**Review by srowen — APPROVED** [review]
I skimmed the changes from 4.9 and didn't see anything that seemed like it was worth investigating. Looks OK for master.

File: python/pyspark/sql/tests.py (hunk 1 above)
Comment (BryanCutler) [inline python/pyspark/sql/tests.py]: ooops, thought I ran local style checks, but maybe it was for scala :(

**Comment by BryanCutler** [comment]
Thanks @holdenk and @srowen for looking into this and reviewing!

**Comment by srowen** [comment]
Yeah I figured you'd need to run `./dev/test-dependencies.sh --replace-manifest` here

**Comment by srowen** [comment]
@BryanCutler if you'll update accordingly I'll merge it.

**Comment by BryanCutler** [comment]
Thanks @srowen , I wasn't aware of that script.  Hopefully this should be good to go now!

**Review by srowen — APPROVED** [review]

**Comment by BryanCutler** [comment]
Hmm, I don't think there should be a MiMa issue right?  It passes locally anyway.. let me try again

**Comment by BryanCutler** [comment]
Jenkins retest this please

**Comment by BryanCutler** [comment]
retest this please

**Comment by srowen** [comment]
Merged to master

**Comment by srowen** [comment]
Merged to 2.0 too
